apps/judicial-system/api/src/app/modules/auth/auth.controller.ts

```diff
@@ -424,34 +456,105 @@ export class AuthController {
     this.redirect(res, currentUser, requestedRedirectRoute)
   }
 
+  private getRedirectRouteForUser(
+    currentUser: User,
+    requestedRedirectRoute?: string,
+  ) {
+    const getRedirectRoute = (
+      defaultRoute: string,
+      allowedPrefixes: string[],
+    ) => {
+      return requestedRedirectRoute &&
+        allowedPrefixes.some((prefix) =>
+          requestedRedirectRoute.startsWith(prefix),
+        )
+        ? requestedRedirectRoute
+        : defaultRoute
+    }
+
+    if (isProsecutionUser(currentUser)) {
+      return getRedirectRoute(CASES_ROUTE, [
+        '/beinir',
```

Comment:
Consider using the routes from the lib, like `CASES_ROUTE`

Reply:
These are route prefixes and as such they do not exists in the lib. However, adding such constants in the lib is a good idea. I would like to clean up all our routes in the future and I think I will leave this as is until then.